crates/wasmtime/src/sync_nostd.rs

```diff
@@ -0,0 +1,249 @@
+//! Synchronization primitives for Wasmtime for `no_std`.
+//!
+//! These primitives are intended for use in `no_std` contexts are are not as
+//! full-featured as the `std` brethren. Namely these panic and/or return an
+//! error on contention. This serves to continue to be correct in the fact of
+//! actual multiple threads, but if a system actually has multiple threads then
+//! something will need to change in the Wasmtime crate to enable the external
+//! system to perform necessary synchronization.
+//!
+//! See a brief overview of this module in `sync_std.rs` as well.
```

Comment:
We could also add hooks to the system interface header to help implement these and push the choice of panicking/aborting or actually doing synchronization out tot he embedder as well down the line. Again, might be worth a comment.

Reply:
Yeah that was my thinking as well in terms of moving it into the header, good point on leaving a comment though